app/src/blocks/blocks.service.ts

```diff
@@ -3,24 +3,40 @@ import { Cron } from "@nestjs/schedule";
 import { Cache } from "cache-manager";
 import { StatusCodes } from "http-status-codes";
 import { BLOCK_GENERATE_INTERVAL, NODE_CACHE_LIMIT } from "src/config";
+import { GatewayService } from "src/gateway/gateway.service";
 import { onChain } from "src/main";
 import { Block } from "src/types/api";
 import { ApiError } from "src/utils/ApiError";
 
 @Injectable()
 export class BlocksService {
-  constructor(@Inject(CACHE_MANAGER) private readonly cacheManager: Cache) {}
+  constructor(
+    @Inject(CACHE_MANAGER) private readonly cacheManager: Cache,
+    @Inject(GatewayService) private readonly gateway: GatewayService
+  ) {}
 
   async onModuleInit() {
     await this.getLatestBlock();
   }
 
-  @Cron(`*/${BLOCK_GENERATE_INTERVAL / 2} * * * * *`)
+  @Cron(`*/${BLOCK_GENERATE_INTERVAL / 2} * * * * *`, {
+    name: "latestBlockSchedule",
+  })
```

Comment:
No way cron is this simple

Reply:
Haha yeah NestJS is turning out to be awesome.